File: src/org/ascore/as/modules/EnumModule.java
package org.ascore.as.modules;

import org.ascore.as.modules.builtins.ModuleBuiltin;
import org.ascore.as.modules.core.ASModuleFactory;
import org.ascore.as.modules.core.ASModuleManager;

public enum EnumModule {
    builtins(ModuleBuiltin::charger),
    ;

    EnumModule(ASModuleFactory moduleFactory) {
        ASModuleManager.enregistrerModule(this, moduleFactory);
    }
}

File: src/org/ascore/as/modules/corekotlin/EnumModule.kt
package org.ascore.`as`.modules.corekotlin

import org.ascore.`as`.modules.EnumModule
import org.ascore.`as`.modules.core.ASModuleFactory
import org.ascore.`as`.modules.core.ASModuleManager

enum class EnumModule(moduleFactory: ASModuleFactory?) {
    ;

    init {
        ASModuleManager.enregistrerModule(EnumModule.valueOf(this.toString()), moduleFactory)
    }
}